scripts/setup-database.sh

```diff
@@ -3,26 +3,76 @@
 source /scripts/env-data.sh
 
 POSTGRES_PASS=$(cat /tmp/PGPASSWORD.txt)
+INITDB_WALDIR_FLAG=""
+
+# Check POSTGRES_INITDB_WALDIR value
+if [[ -n "${POSTGRES_INITDB_WALDIR}" ]]; then
+    # If POSTGRES_INITDB_WALDIR is defined, make sure that it is not inside 
+    # the ${DATADIR} directory, to avoid deletions
+    case "${POSTGRES_INITDB_WALDIR}" in
+        ${DATADIR}/*)
```

Comment:
We use pattern match to check if `POSTGRES_INITDB_WALDIR` is a subpath of ${DATADIR}